src/hooks/useDeployment.ts

```diff
@@ -0,0 +1,98 @@
+import { useSession } from "next-auth/react";
+import { useState, useCallback, useMemo } from "react";
+
+import { MaciService } from "~/services/maci";
+import { useEthersSigner } from "./useEthersSigner";
+import { eas } from "~/config";
+import * as vks from "../../public/vk_10-2-1-2_test.0.json";
+
+interface IDeployArgs {
+  amount: number;
+}
+
+export function useDeployment() {
+  /**
+   * deployStatus:
+   * 0. default
+   * 1. VoiceCreditProxy
+   * 2. Gatekeeper
+   * 3. Verifier
+   * 4. PollFactory
+   * 5. MessageProcessorFactory
+   * 6. TallyFactory
+   * 7. MACI contract
+   * 8. VkRegistry
+   */
+  const [deployStatus, setDeployStatus] = useState<number>(0);
+  const signer = useEthersSigner();
+  const { data } = useSession();
+
+  const maci = useMemo(
+    () => (signer ? new MaciService(signer) : undefined),
+    [Boolean(signer), useEthersSigner],
+  );
+
+  const deploy = useCallback(
```

Comment:
should we add error handling?

Reply:
resolve this by adding simple try and catch now, would make it more complete in the wizard PR.